### PR TITLE
SDK const migration

### PR DESCRIPTION
**Pull request by jmrossy**
- Finish migration of SDK consts started in #3 
- Convert json chain files to yaml
- Create build script to transform data
- Set up CI job and mocha tests
- Productize package.json file
- Update Readme
- Lock yarn version like in monorepo

### CODE REVIEW

File: chains/polygon/addresses.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+aggregationHook: "0x34dAb05650Cf590088bA18aF9d597f3e081bCc47"
```
Comment (yorhodes) [inline chains/polygon/addresses.yaml:1]: should we omit the quotes?

File: scripts/migrate-sdk-consts.js
Comment (yorhodes) [inline scripts/migrate-sdk-consts.js:1]: nice
should we remove this or keep jic?

File: scripts/update-chain-schema.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import fs from 'fs'
+import { zodToJsonSchema } from "zod-to-json-schema";
+import {ChainMetadataSchemaObject} from "@hyperlane-xyz/sdk";
+
+const schema = zodToJsonSchema(ChainMetadataSchemaObject, "hyperlaneChainMetadata");
```
Comment (yorhodes) [inline scripts/update-chain-schema.js:5]: this is cool!

File: chains/schema
Comment (yorhodes) [inline chains/schema:1]: should we git ignore this if its generated?

File: scripts/build.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+function genJsExport(data, exportName) {
+  return `export const ${exportName} = ${JSON.stringify(data, null, 2)};`;
+}
+
+let chainMetadata = {};
+let chainAddresses = {};
+
+console.log('Parsing and copying chain data');
+for (const file of fs.readdirSync('./chains')) {
```
Comment (yorhodes) [inline scripts/build.ts:14]: what about the core.ts and schema files?

File: scripts/build.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  const inDirPath = `./chains/${file}`;
+  const assetOutPath = `./dist/chains/${file}`;
+  // generated ts files to to the tmp dir so they can be compiled along with other generated code
+  const tsOutPath = `./tmp/chains/${file}`;
+  const stat = fs.statSync(`${inDirPath}`);
+  if (!stat.isDirectory()) continue;
```
Comment (yorhodes) [inline scripts/build.ts:20]: ah I see

File: scripts/build.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import { pick } from '@hyperlane-xyz/utils';
+import fs from 'fs';
+import { parse } from 'yaml';
+import { CoreChains } from '../chains/core';
+
+function genJsExport(data, exportName) {
+  return `export const ${exportName} = ${JSON.stringify(data, null, 2)};`;
+}
```
Comment (yorhodes) [inline scripts/build.ts:8]: is this a common pattern? kinda surprised that we are exporting non-static assets tbh

Comment (yorhodes) [reply scripts/build.ts:8]: how useful is this?

File: scripts/build.ts
Comment (yorhodes) [inline scripts/build.ts:1]: slightly worried this is fragile but love the backward compatibility

File: test/chains.test.ts
Comment (yorhodes) [inline test/chains.test.ts:1]: nice!

Comment (yorhodes) [reply test/chains.test.ts:1]: could we adapt this to use the bundled exports and skip the readAllFiles util?

File: test/chains.test.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { ChainMetadataSchema } from '@hyperlane-xyz/sdk';
+import { z } from 'zod';
+import { readAllChainYamlFiles } from './utils';
+
+describe('Chain metadata', () => {
+  it('All chains have valid metadata', () => {
+    const result = readAllChainYamlFiles('metadata.yaml');
+    for (const [chain, metadata] of Object.entries(result)) {
+      console.log('Validating metadata for', chain);
+      ChainMetadataSchema.parse(metadata);
```
Comment (yorhodes) [inline test/chains.test.ts:10]: nit but slightly prefer separate 
```it(`${chain} metadata is valid`, () =>```

**Review by yorhodes — APPROVED** [review]
this looks great

File: chains/polygon/addresses.yaml (hunk 1 above)
Comment (jmrossy) [reply chains/polygon/addresses.yaml:1]: I believe yaml may consider it a number. I didn't put them in explicitly, this is just what the yaml lib spit out

File: scripts/migrate-sdk-consts.js
Comment (jmrossy) [reply scripts/migrate-sdk-consts.js:1]: I'd recommend we keep jib for a week or two. I'll take the AI to remove it when things are settled.

File: scripts/build.ts (hunk 3 above)
Comment (jmrossy) [reply scripts/build.ts:14]: core.ts gets copied over into index.ts which then gets compiled by ts.
The schema I forgot, I'll get that included too

File: scripts/build.ts
Comment (jmrossy) [reply scripts/build.ts:1]: I feel good about it personally. It's simple stuff even if we need to overhaul

File: test/chains.test.ts
Comment (jmrossy) [reply test/chains.test.ts:1]: Oh maybe, I'll give it a try now

File: test/chains.test.ts (hunk 6 above)
Comment (jmrossy) [reply test/chains.test.ts:10]: You got it 👍

File: test/chains.test.ts
Comment (jmrossy) [reply test/chains.test.ts:1]: Done, good idea

**Pull request merged**